File: contests/MRC-Qual-2025/B.cpp
// Задача: B, Вердикт: OK, Время: 2ms, Память: 844.00Kb
#include <bits/stdc++.h>
using namespace std;
main() {
    int birthYear;
    cin >> birthYear;
    int joinYear;
    cin >> joinYear;
    int numSems;
    cin >> numSems;
    int numFinals;
    cin >> numFinals;
    int numRegionals;
    cin >> numRegionals;
    // определяем ответ
    string answer = "+"; // изначально считаем, что участие разрешено, далее проверки
    if (numFinals >= 2) {
        answer = "-"; // участие запрещено
    } else if (numRegionals >= 6) {
        answer = "-"; // участие запрещено
    } else if (numRegionals == 5 || (birthYear < 2002 && joinYear < 2021)) {
        // применяется правило 5
        if (numSems >= 9) // участие запрещено
            answer = "-";
        else // участие под вопросом
            answer = "?";
    }
    cout << answer << endl;
}

File: contests/MRC-Qual-2025/B.py
# Задача: B, Вердикт: OK, Время: 113ms, Память: 32.36Mb
import sys

def main():
    # читаем вход целыми числами
    data = list(map(int, sys.stdin.read().strip().split()))
    birthYear, joinYear, numSems, numFinals, numRegionals = data
    # определяем ответ
    answer = "+"  # изначально считаем, что участие разрешено, далее проверки
    if numFinals >= 2:
        answer = "-"  # участие запрещено
    elif numRegionals >= 6:
        answer = "-"  # участие запрещено
    elif numRegionals == 5 or (birthYear < 2002 and joinYear < 2021):
        # применяется правило 5
        if numSems >= 9:  # участие запрещено
            answer = "-"
        else:  # участие под вопросом
            answer = "?"
    print(answer)

if __name__ == "__main__":
    main()
